composer/algorithms/mixup/mixup.py

```diff
@@ -127,46 +118,67 @@ class MixUp(Algorithm):
             )
     """
 
-    def __init__(self, num_classes: int, alpha: float = 0.2):
-        self.num_classes = num_classes
+    def __init__(self, alpha: float = 0.2, interpolate_loss: bool = True):
         self.alpha = alpha
+        self.interpolate_loss = interpolate_loss
         self.mixing = 0.0
         self.indices = torch.Tensor()
+        self.permuted_target = torch.Tensor()
+        self._loss_fn = None  # set on Event.INIT
 
     def match(self, event: Event, state: State) -> bool:
-        """Runs on Event.INIT and Event.AFTER_DATALOADER.
-
-        Args:
-            event (:class:`Event`): The current event.
-            state (:class:`State`): The current state.
-        Returns:
-            bool: True if this algorithm should run now.
-        """
-        return event == Event.AFTER_DATALOADER
+        if self.interpolate_loss:
+            return event in [Event.INIT, Event.AFTER_DATALOADER, Event.AFTER_LOSS]
+        else:
+            return event in [Event.AFTER_DATALOADER, Event.BEFORE_LOSS]
 
     def apply(self, event: Event, state: State, logger: Logger) -> None:
-        """Applies MixUp augmentation on State input.
-
-        Args:
-            event (Event): the current event
-            state (State): the current trainer state
-            logger (Logger): the training logger
-        """
-
-        input, target = state.batch_pair
-        assert isinstance(input, torch.Tensor) and isinstance(target, torch.Tensor), \
-            "Multiple tensors for inputs or targets not supported yet."
-
-        self.mixing = _gen_mixing_coef(self.alpha)
-
-        new_input, new_target, self.indices = mixup_batch(
-            input,
-            target,
-            mixing=self.mixing,
-            num_classes=self.num_classes,
-        )
+        if event == Event.INIT:
+            # Grab the loss function
+            if isinstance(state.model, ComposerModel):
+                self._loss_fn = state.model.loss
+            else:
+                raise RuntimeError("Model must be of type ComposerModel")
+
+        if event == Event.AFTER_DATALOADER:
+            input, target = state.batch_pair
+            assert isinstance(input, torch.Tensor) and isinstance(target, torch.Tensor), \
+                "Multiple tensors for inputs or targets not supported yet."
```

Comment:
`raise NotImplementedError` instead of `assert`-ing

Reply:
Fixed, and broke this up into two so the error message will identify whether `input` or `target` was not as expected.